servicetalk-encoding-netty/src/main/java/io/servicetalk/encoding/netty/AbstractContentCodec.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © 2021 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.encoding.netty;
+
+import io.servicetalk.encoding.api.ContentCodec;
+
+import java.util.Objects;
+
+abstract class AbstractContentCodec implements ContentCodec {
+
+    private final CharSequence name;
+
+    AbstractContentCodec(final CharSequence name) {
+        this.name = name;
+    }
+
+    public final CharSequence name() {
+        return name;
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
```

Comment:
To capture @bondolo comment here, see https://github.com/apple/servicetalk/pull/1383/files#r583080915:

> comparing the exact class and instanceof is not quite the same. The existing implementation would declare different sub-classes to be non-equal where this implementation might declare them equivalent. There is also the danger of non-reflexive equals() if one of the sub-classes declares an equals() method that matches specifically against their own class.
I would encourage you to leave it getClass() != o.getClass() unless you can make the equals() method final.

@tkountis IIUC, for the purpose of content codecs it doesn't matter if 2 codecs have the same class or not. the only `name` matters. the `instanceof` probably makes more sense here, but should we make this `equals` and `hashCode` impl `final`?

Reply:
Correct. Made them final and changed it to instanceof